### ISSUE TITLE
Schedule meeting feature implementation

### ISSUE THREAD

**Issue by evilrat** [evilrat/flutter_zoom_sdk#45]
@evilrat , It would be great if you can also plan the Schedule Meeting feature. Like if the host wants to schedule the meeting, after scheduling the meeting, he will get the scheduled meeting details.

_Originally posted by @Achin-verma in https://github.com/evilrat/flutter_zoom_sdk/issues/27#issuecomment-990935684_

**Comment by itsAyyazdev** [comment]
any update on schedule meetings?

**Comment by Achin-verma** [comment]
Yes, for this feature, we need to use zoom meetings APIs. For the schedule and join as a host and join as participants. You have to follow this flow. 

1. First, you have to buy the zoom user license, which is paid.
2. Then from the zoom API, you have to use this API to create the meeting. [https://marketplace.zoom.us/docs/api-reference/zoom-api/meetings/meetingcreate](url).
3. Now, you have to save this response on your server and this response will give you all the details to join as a host and join participants
4. To join as participants, you need a meeting id and meeting password, from the above api response. Which you have to pass in this zoom sdk.
5. To join as a host, you need, a meeting id, licensed user email as user id, display name, zak, zoom access token, you can get it from start_url from creating meeting API response. I am posting the code to join as a host via using the android zoom sdk,

To Join as a Host.
```
public int startMeetingWithHost() {
    ZoomSDK mZoomSDK = ZoomSDK.getInstance();
    if(!mZoomSDK.isInitialized()) {
      System.out.println("==========Not initialized!!!!!!=====");
      return -1;
    }

    int ret = -1;
    MeetingService meetingService = mZoomSDK.getMeetingService();
    if(meetingService == null) {
      return ret;
    }

    StartMeetingOptions opts = getStartMeetingOptions();
    opts.no_driving_mode = true;
    opts.no_invite = true;
    opts.meeting_views_options = MeetingViewsOptions.NO_TEXT_PASSWORD + MeetingViewsOptions.NO_TEXT_MEETING_ID;

    StartMeetingParamsWithoutLogin params = new StartMeetingParamsWithoutLogin();
    params.userId = userId;
    params.userType = MeetingService.USER_TYPE_SSO;
    params.displayName = displayName;
    params.zoomAccessToken = zak;
    params.meetingNo = meetingNo;
    ret = meetingService.startMeetingWithParams(context, params, opts);
    Log.i("startMeetingWithNumber", "======startMeetingWithNumber=====ret=" + ret);
    if(ret == MeetingError.MEETING_ERROR_SUCCESS){
      
    }else{
      MEETING ERROR
    }
    return ret;
  }

  public static StartMeetingOptions getStartMeetingOptions() {
    StartMeetingOptions opts = new StartMeetingOptions();
    return opts;
  }
```

**Comment by evilrat** [comment]
> 2\. https://marketplace.zoom.us/docs/api-reference/zoom-api/meetings/meetingcreate

i don't think that users will buy license for zoom...if they can alternatively schedule meeting

**Comment by evilrat** [comment]
> 2\. [https://marketplace.zoom.us/docs/api-reference/zoom-api/meetings/meetingcreate](url).

above link has api and provide some response......and this plugin is used only by developers so can can call api using dio and save response and can use it later as per their need......so i don't think their is any use of providing this feature in this plugin........if you still think that this needs to be there in plugin...then we will look into it...and try to implement it as soon as possible.....thanks

**Comment by selvesandev** [comment]
Hi @evilrat 

Yes this will be required. As we are not being able to host meeting with the zoom's API user. These user don't have credentials. They are allowed to start meeting with ZAK token and the participants can participate. 
Using this library `flutter_zoom_plugin` we could do so. https://github.com/decodedhealth/flutter_zoom_plugin/blob/88d1fa2685f28d67410d46e094461a0a030d3192/lib/zoom_view.dart#L83-L84

**Comment by evilrat** [comment]
> Hi @evilrat
> 
> Yes this will be required. As we are not being able to host meeting with the zoom's API user. These user don't have password but can still start meeting with `ZAK token`, `meeting id` , `user id` and `meeting password` and the participants can participate. This library `flutter_zoom_plugin` had that feature implemented. https://github.com/decodedhealth/flutter_zoom_plugin/blob/88d1fa2685f28d67410d46e094461a0a030d3192/lib/zoom_view.dart#L83-L84

Schedule meeting with your personal meeting id and password.....and use that to join meeting and start meetig

**Comment by selvesandev** [comment]
Yep we can totally do that but when there is multiple user hosting meeting at the same time (i.e each api user hosting their own personal meeting at the same time and each have different set of participants)

**Comment by evilrat** [comment]
> Yep we can totally do that but when there is multiple user hosting meeting at the same time (i.e each api user hosting their own personal meeting concurrently and each have different set of participants). This is not working for this scenario

i get it

**Comment by dery86** [comment]
@evilrat is there any update about when we could make the user join as a host?
